### ISSUE TITLE
Mode3 suggestion

### ISSUE THREAD

**Issue by boris1978** [Courseplay/courseplay#2747]
Please change in file mode3.lua line 26

from
courseplay:setCustomTimer(vehicle, "fillLevelChange", 10);

to
courseplay:setCustomTimer(vehicle, "fillLevelChange", 15);

I like to play with 2 or 3 transport trailers and 2 or 3 chaser bins. When chaser bin is unloading and first transport trailer is full, 10 sec is too short for second trailer to move under the auger pipe, it needs at least 15 sec.

**Comment by AndyK70** [comment]
maybe it would be more convenient if this timer delay was a variable in **courseplaySettings.xml**?

I'll have a look and if I got this I'll do a pull request.

**Comment by pvajko** [comment]
@boris1978 , @AndyK70 thank you, we really appreciate if an issue/suggestion comes with a solution, especially in the form of a pull request.

**Comment by AndyK70** [comment]
@boris1978 please download the zip from this branch:
[https://github.com/AndyK70/courseplay/tree/fillLevelChangeDelay](https://github.com/AndyK70/courseplay/tree/fillLevelChangeDelay)

Extract it and everything **inside** the folder **courseplay-fillLevelChangeDelay** zip to new archive called FS17_CoursePlay5.zip then place it in your mod folder (like you do with normal developer builds).
After saving your savegame once open the vehicle.xml with a text editor and you'll find the variable fillLevelChangeDelay where you can change it to your desired delay in seconds.

Please test it and report back if it worked out for you or if you ran into any issues.

**Comment by pops64** [comment]
You could also adjust the drive on precent to avoid this in the first place. Yes we know that it is a shared variable atm which is something that has been on my to-do list to fix

**Comment by boris1978** [comment]
@AndyK70  thanks I will test it and will write report, but if I need to make change in .xml file with text editor it is not a better solution for me because I have done that with .lua file before I have opened this suggestion.
If developer change basic value from 10 to 15 it is good enough for me, because I don't need to make change of that value anymore.

**Comment by AndyK70** [comment]
In my opinion it is a difference if you are going to change a value in the script itself and breaking the checksum and the ingame update process, or adjusting a variable in a config file.
And don't overlook that this can be only the first step to an addition in the mode3 HUD, to set this value ingame.

**Comment by boris1978** [comment]
@AndyK70  Yes I agree with you.
I've changed values of vehicles.xml file from 10 to 20 at 3 lines
line 124
line 143
line 162
With this change it works perfect for me. I do not play multiplayer.
At first I've changed value only at line 124 and it didn't work. All that three lines have aimode="3"

**Comment by AndyK70** [comment]
@boris1978 your vehicles.xml is not my vehicles.xml so line numbers don't have any meaning :wink:
The new variable fillLevelChangeDelay is added to all vehicles (combines, tractors, semis), so you'll have to look up that node in the xml for which vehicle you want to change it. 
It is not a global variable as I intended first but is set to any vehicle specifically. So you can have one tractor having it set to 5s and another having it set to 20s or more if you desire.

The aimode="3" value is only a snapshot and can change at any time if the corresponding tractor does another job, e.g. sowing. Don't rely on that, better look for the specific vehicle you want to change it for.

**Comment by pops64** [comment]
No. There is CP timer that is 10 secs long and starts after no fill level change is dectected in the tipper and the drive on precent is lower than the set vaule provide by the HUD. After 10 secs the tipper moves on. Back to my post if you change the drive on precent to 0 this condition can never be met and so it won't ever drive on. The problem is it is shared variable with the havesters call function. Which is something I have been meaning to fix

**Comment by boris1978** [comment]
@AndyK70  Sorry I was a little bit in hurry and I'm not softwere developer. Yes I now understand that I need to look for vehicle id. It will be great to have option in the HUD to change this value when vehicle is working in mode3, like we do with "field speed", "max speed" for example. I did not have any issues with CP or with specific mode, but it was only 1 hour of gaming. I will play FS17 with CP from your link, if I found any issues I will report.
